src/lando/main/models/commit_map.py

```diff
@@ -0,0 +1,90 @@
+import requests
+from django.db import models
+
+from lando.main.models import BaseModel
+from lando.main.models.repo import Repo
+
+
+class CommitMap(BaseModel):
+    """Map a git hash to an hg hash, based on a specific repo."""
+
+    HGMO_PUSHLOG_TEMPLATE = "https://hg.mozilla.org/{}/json-pushes"
+    REPO_MAPPING = (("firefox", "mozilla-unified"),)
+
+    git_hash = models.CharField(default="", unique=True)
+    hg_hash = models.CharField(default="", unique=True)
+
+    # NOTE: This value is set because multiple Lando repos can map to a single hg repo.
+    # This is because currently a separate repo object is created for each branch of a
+    # git repo, though those might map to a single hg repo (e.g., mozilla-unified).
+    git_repo_name = models.CharField(default="")
+
+    @classmethod
+    def get_hg_repo_name(cls, git_repo_name: str) -> str:
+        """Return mapped repo name or `git_repo_name` by default."""
+        return dict(cls.REPO_MAPPING).get(git_repo_name, git_repo_name)
+
+    @classmethod
+    def get_git_repo_name(cls, hg_repo_name: str) -> str:
+        """Return mapped repo name or `hg_repo_name` by default."""
+        return {hg: git for git, hg in cls.REPO_MAPPING}.get(hg_repo_name, hg_repo_name)
+
+    @classmethod
+    def get_pushlog_url(cls, git_repo_name: str) -> str:
+        """Return pushlog URL based on provided repo name."""
+        return cls.HGMO_PUSHLOG_TEMPLATE.format(cls.get_hg_repo_name(git_repo_name))
+
+    @classmethod
+    def _find_last_node(cls, git_repo_name: str) -> "CommitMap":
+        """Return last CommitMap object that was stored for given repo."""
+        return cls.objects.filter(git_repo_name=git_repo_name).latest("created_at")
```

Comment:
nit: this is the creation date in the DB, rather than than of the commit, and this is loosely related to the topological order of the commits depending on how the insertion is done.

I had the same issue in the PushLog, and decided that it was fine, but it's worth keeping in mind and making the assumption explicitely.